packages/flutter/lib/src/widgets/inherited_notifier.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'framework.dart';
+
+/// An inherited widget for a [Listenable] [notifier], which updates its
+/// dependencies when the [notifier] is triggered.
+///
+/// This is a variant of [InheritedWidget], specialised for subclasses of
+/// [Listenable], such as [ChangeNotified] or [ValueNotifier].
+///
+/// Dependents are notified whenever the [notifier] sends notifications, or
```

Comment:
This is a little confusing because Listenable isn't (necessarily) a Notifier and the developer might be wondering what "sends notifications" means.